python_heideltime/python_heideltime.py

```diff
@@ -16,76 +16,118 @@
+    def set_document_time(self, document_time: str):
+        """
+        Expects a string in the form of 'YYYY-MM-DD' to specify the document creation time (DCT).
+        Only used in combination with 'NEWS' or 'COLLOQUIAL' document types.
+        """
+        if not regex.match(r'[0-9]{4}-[0-9]{2}-[0-9]{2}', document_time):
+            raise ValueError('Incorrect format for document time specified. Please use the \"YYYY-MM-DD\" format.')
         self.document_time = document_time
 
-    def set_language(self, language):
-        self.language = language
+    def set_language(self, language: str):
+        """
+        Sets the language of the parser. Available languages according to Heideltime docs are:
+        ENGLISH, GERMAN, DUTCH, ENGLISHCOLL (for -t COLLOQUIAL), ENGLISHSCI (for -t SCIENTIFIC), SPANISH, ITALIAN,
+        ARABIC, VIETNAMESE, FRENCH, CHINESE, RUSSIAN, CROATIAN, PORTUGUESE, ESTONIAN.
+        """
+        if not language.upper() in AVAILABLE_LANGUAGES:
+            raise ValueError(f'Unknown language specified! Please specify one of the supported languages below:\n'
+                             f'{AVAILABLE_LANGUAGES}')
+        self.language = language.upper()
 
-    # called Type in HeidelTime
-    def set_document_type(self, doc_type):
+    def set_document_type(self, doc_type: str):
+        """
+        Sets the document type. Can be either of 'NARRATIVES', 'NEWS', 'COLLOQUIAL', or 'SCIENTIFIC'.
+        """
+        if not doc_type.upper() in AVAILABLE_DOCUMENT_TYPES:
+            raise ValueError(f'Unknown document type specified! Please use one of the following supported document'
+                             f'types: {AVAILABLE_DOCUMENT_TYPES}')
         self.doc_type = doc_type
 
     def set_output_type(self, output_type):
+        """
+        Set the output type of the parser. Heideltime supports either of 'XMI' or 'TIMEML', and defaults to the latter.
+        """
+        if not output_type.upper() in AVAILABLE_OUTPUT_TYPES:
+            raise ValueError(f'Unknown output type specified! Plese use one of the following supported output types:'
+                             f'{AVAILABLE_OUTPUT_TYPES}')
         self.output_type = output_type
 
     def set_encoding(self, encoding):
+        """
+        Set the corresponding output encoding used by Heideltime. It is unclear from the original docs,
+        which exact encodings are supported.
+        """
         self.encoding = encoding
 
-    # this needs a full path
-    def set_config_file(self, config_file):
+    def set_config_file(self, config_file: str):
+        """
+        Set the path of Heideltime config file. Requires a full (absolute) path.
+        """
         self.config_file = config_file
 
-    # True / False
-    def set_verbosity(self, verbosity):
+    # FIXME: Technically, Heideltime supports two different verbosity levels, which we cannot model here.
```

Comment:
Maybe specify here what you mean with that. I would not remember what I meant here a month alter. ;)